flexmeasures/data/schemas/times.py

```diff
@@ -15,6 +15,42 @@ class DurationValidationError(FMValidationError):
     status = "INVALID_PERIOD"  # USEF error status
 
 
+class NewDurationField(MarshmallowClickMixin, fields.Str):
+    """Field that deserializes to a ISO8601 Duration
+    and serializes back to a string."""
+
+    def _deserialize(self, value, attr, obj, **kwargs) -> str:
+        """
+        Use the isodate library to turn an ISO8601 string into a timedelta.
+        For some non-obvious cases, it will become an isodate.Duration, see
+        ground_from for more.
+        This method throws a ValidationError if the string is not ISO norm.
+        """
+        try:
+            value_isodate = isodate.parse_duration(value)
+        except ISO8601Error as iso_err:
+            raise DurationValidationError(
+                f"Cannot parse {value} as ISO8601 duration: {iso_err}"
+            )
+
+        if value_isodate.seconds % 60 != 0 or value_isodate.microseconds != 0:
```

Comment:
We should mention this in the docstring

Reply:
Mentioned the check for whole minutes in the docstring